### ISSUE TITLE
Some crashes in addons manager

### ISSUE THREAD

**Issue by deveee** [supertuxkart/stk-code#3002]
There are few different crashes, but the reason may be similar.

```
  native: pc 000000000004ad44  /system/lib/libc.so (tgkill+12)
  native: pc 00000000000484e3  /system/lib/libc.so (pthread_kill+34)
  native: pc 000000000001d495  /system/lib/libc.so (raise+10)
  native: pc 0000000000018fd1  /system/lib/libc.so (__libc_android_abort+34)
  native: pc 0000000000017034  /system/lib/libc.so (abort+4)
  native: pc 0000000000aa9771  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZN9__gnu_cxx27__verbose_terminate_handlerEv+228)
  native: pc 0000000000a72abd  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZN10__cxxabiv111__terminateEPFvvE+4)
  native: pc 0000000000a72b31  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZSt9terminatev+8)
  native: pc 0000000000a73127  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (__cxa_pure_virtual+14)
  native: pc 00000000005ee1e1  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZNK3irr2io11CFileSystem9existFileERKNS_4core6stringIcNS2_12irrAllocatorIcEEEE+32)
  native: pc 000000000028e16f  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZNK11FileManager10fileExistsERKSs+78)
  native: pc 00000000003b9265  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZN13AddonsManager13downloadIconsEv+412)
  native: pc 00000000003bea6b  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZN13AddonsManager10initAddonsEPK7XMLNode+4478)
  native: pc 00000000003bf09d  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZN13AddonsManager4initEPK7XMLNodeb+424)
  native: pc 00000000003b6bf9  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZN11NewsManager12downloadNewsEPv+376)
  native: pc 0000000000047fb3  /system/lib/libc.so (_ZL15__pthread_startPv+22)
  native: pc 0000000000019a1d  /system/lib/libc.so (__start_thread+6)
```

It crashed in ```lib/irrlicht/include/irrString.h:355``` 

```
  native: pc 0000000000044310  /system/lib/libc.so (tgkill+12)
  native: pc 0000000000041f11  /system/lib/libc.so (pthread_kill+32)
  native: pc 000000000001ba13  /system/lib/libc.so (raise+10)
  native: pc 0000000000018c81  /system/lib/libc.so (__libc_android_abort+34)
  native: pc 0000000000016840  /system/lib/libc.so (abort+4)
  native: pc 000000000001a627  /system/lib/libc.so (__libc_fatal+16)
  native: pc 000000000002e221  /system/lib/libc.so (__bionic_heap_corruption_error+8)
  native: pc 000000000003048d  /system/lib/libc.so (dlfree+308)
  native: pc 00000000003c1e0b  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZN13AddonsManagerD1Ev+298)
  native: pc 0000000000242f17  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (main+2366)
  native: pc 000000000027b8f7  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (android_main+42)
  native: pc 00000000009f065b  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so
  native: pc 0000000000041813  /system/lib/libc.so (_ZL15__pthread_startPv+30) 
  native: pc 0000000000019315  /system/lib/libc.so (__start_thread+6)
```

Crashed in ```lib/irrlicht/include/irrAllocator.h:41```

```
  native: pc 000000000028f65c  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZNK11FileManager10fileExistsERKSs+55)
  native: pc 0000000000291be9  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZNK11FileManager10removeFileERKSs+16)
  native: pc 000000000055a67d  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZN6Online11HTTPRequest9operationEv+588)
  native: pc 000000000055550b  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZN6Online7Request7executeEv+86)
  native: pc 0000000000552da3  /data/app/org.supertuxkart.stk-1/lib/arm/libmain.so (_ZN6Online14RequestManager8mainLoopEPv+122)
  native: pc 0000000000047483  /system/lib/libc.so (_ZL15__pthread_startPv+22) 
  native: pc 0000000000019dcd  /system/lib/libc.so (__start_thread+6)
```

Crashed in ```lib/irrlicht/include/irrString.h:391```

It looks that the first crash is a "pure virtual method called" crash. And I sometimes see it on linux too.

**Comment by Benau** [comment]
a77b897cd5b73d1833a32dfa227d37a54d8bafb6 good for u?

And where is the fix for isPlaying music in #2169 and #3000, would be good for next beta apk......

**Comment by deveee** [comment]
I was thinking about something similar and indeed it seems to have a sense. For now I assume that it's fixed and we will see crash reports after next beta.

And I don't think that I will fix #3000. The proper fix is to make sure that all global variables are initialized on startup. But even if I will do it, I can't assume that noone will break it later. Or I can make dummy libmain.so that only loads real libstk.so, but it's too much of work and little gain and I can't assume that I will always get correct path to .so libraries (I don't trust it enough).

**Comment by Benau** [comment]
and #2169? (or i should do it for you?)
